src/snowflake/connector/crl.py

```diff
@@ -0,0 +1,586 @@
+                    crl.tbs_certlist_bytes,
+                    hash_algorithm,
+                )
+
+            logger.debug("CRL signature verification successful")
+            return True
+        except Exception as e:
+            logger.warning("CRL signature verification failed: %s", e)
+            return False
+
+    def _check_certificate_against_crl(
+        self, cert: x509.Certificate, crl: x509.CertificateRevocationList
+    ) -> CRLValidationResult:
+        """Check if certificate is revoked according to CRL"""
+        revoked_cert = crl.get_revoked_certificate_by_serial_number(cert.serial_number)
+        return (
+            CRLValidationResult.REVOKED
+            if revoked_cert
+            else CRLValidationResult.UNREVOKED
+        )
+
+    def validate_connection(self, connection: SSLConnection | SSLObject) -> bool:
+        """
+        Validate an connection against CRLs.
+
+        This method extracts certificate chains from the connection and validates them
+        against Certificate Revocation Lists (CRLs).
+
+        Args:
+            connection: OpenSSL or SSLObject connection object
+
+        Returns:
+            True if validation passes, False otherwise
+        """
+        try:
+            if isinstance(connection, SSLObject):
+                # AIOHTTP SSLObject case
+                cert_chain = connection._sslobj.get_unverified_chain()
+                cert_der_chain = [
+                    ssl.PEM_cert_to_DER_cert(c.public_bytes()) for c in cert_chain
+                ]
+            else:
+                # OpenSSL SSLConnection case
+                cert_chain = connection.get_peer_cert_chain()
+                cert_der_chain = [
+                    dump_certificate(FILETYPE_ASN1, c) for c in cert_chain
+                ]
+
+            certificate_chains = self._extract_certificate_chains_from_connection(
+                cert_der_chain
+            )
+        except Exception as e:
+            logger.warning(
+                "Failed to extract certificate chain for CRL validation: %s", e
+            )
+            certificate_chains = []
+        return self.validate_certificate_chains(certificate_chains)
+
+    def _extract_certificate_chains_from_connection(
+        self, cert_chain: list[bytes]
+    ) -> list[list[x509.Certificate]]:
+        """Extract certificate chains from list of certs.
+
+        Args:
+            cert_chain: List of DER-encoded certificates from the connection
+
+        Returns:
+            List of certificate chains, where each chain is a list of x509.Certificate objects
+        """
+
+        if not cert_chain:
+            logger.debug("No certificate chain found in connection")
+            return []
+
+        # Convert OpenSSL certificates to cryptography x509 certificates
+        x509_chain = []
+        for cert_der in cert_chain:
+            cert_x509 = x509.load_der_x509_certificate(cert_der, default_backend())
```

Comment:
On main branch there's also cert to bytes conversion. But as it differs for aio and sync scenarios I moved it to earlier stage.